clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6365,6 +6365,28 @@ void OffloadBundler::ConstructJob(Compilation &C, const JobAction &JA,
       Triples += CurDep->getOffloadingArch();
     }
   }
+
+  // When bundling for FPGA with -fsycl-link a specific triple is formulated
+  // to match the FPGA binary.  We are also guaranteed only the single device
+  // and host object inputs
+  const ToolChain *TCCheck = &getToolChain();
+  if (TCArgs.hasArg(options::OPT_fsycl_link_EQ) &&
+      TCCheck->getTriple().getSubArch() == llvm::Triple::SPIRSubArch_fpga) {
+    Triples = "-targets=";
+    llvm::Triple TT;
+    TT.setArchName(JA.getInputs()[0]->getType() == types::TY_AOCX ? "fpga_aocx"
+                                                                 : "fpga_aocr");
+    TT.setVendorName("intel");
+    TT.setOS(llvm::Triple(llvm::sys::getProcessTriple()).getOS());
```

Comment:
AFAIU, we can do a cross compilation, so should not the OS component from the from the TCCheck->getTriple() be used?

Reply:
makes since - will adjust.